src/pages/Models/Details/generators.js

```diff
@@ -409,6 +412,27 @@ export function generateMachineRows(
     return [];
   }
 
+  const generateMachineApps = (machineId) => {
+    const appsOnMachine = [];
+    const modelApplications = modelStatusData?.applications;
+    modelApplications &&
+      Object.entries(modelApplications).forEach(([appName, appInfo]) => {
+        appInfo?.units &&
+          Object.values(appInfo.units).forEach((unitInfo) => {
+            if (machineId === unitInfo.machine) {
+              appsOnMachine.push([appName, appInfo.charm]);
+            }
+          });
+      });
+
+    const apps = appsOnMachine.length
+      ? appsOnMachine.map((app) => {
+          return generateIconImg(app[0], app[1]);
+        })
+      : "There are no apps on this machine";
```

Comment:
This is a bit too long and wraps, maybe just...
```suggestion
      : "none";
```
or
```suggestion
      : "";
```

![Screen Shot 2021-01-12 at 11 51 10 AM](https://user-images.githubusercontent.com/532033/104352604-7e106f80-54cc-11eb-81a1-a1361b7700f1.png)

Reply:
That's not the design?